t/integration/tasks.py

```diff
@@ -41,3 +41,20 @@ def cumadd(requests: List[SimpleRequest]) -> None:
     for request in requests:
         result += request.args[0]
         current_app.backend.mark_as_done(request.id, result, request=request)
+
+
+@shared_task(base=Batches, flush_every=2, flush_interval=1)
+def retry_if_even(requests: List[SimpleRequest]) -> None:
+    """Retry the task if the first argument of a request is even."""
+    from celery import current_app
+
+    for request in requests:
+        if request.args[0] % 2 == 1:
+            # Odd, success
+            current_app.backend.mark_as_done(request.id, True, request=request)
+        else:
+            # Even, so modify to be odd next time around and retry
+            request.args = (request.args[0] + 1,)
+            retry_if_even.apply_async(
+                args=request.args, kwargs=request.kwargs, countdown=3
+            )
```

Comment:
It feels weird/unnecessary to modify the incoming `SimpleRequest` object although there isn't really anything wrong with it.

```suggestion
            # Even, modify it to be odd next time around and retry.
            retry_if_even.apply_async(
                args=(request.args[0] + 1,), kwargs=request.kwargs, countdown=3
            )
```